src/vmm/src/guest_config/mod.rs

```diff
@@ -1,5 +1,26 @@
 // Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+use serde::{Deserialize, Serialize};
+
 #[cfg(cpuid)]
 pub mod cpuid;
+
+/// Module containing type implementations needed for aarch64 (ARM) CPU configuration
+#[cfg(target_arch = "aarch64")]
+pub mod aarch;
+
+/// Module containing type implementations needed for x86 CPU configuration
+#[cfg(target_arch = "x86_64")]
+pub mod x86;
+
+/// Bit-mask based wrapper to apply
+/// changes to a given register's value
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct Modifier {
+    /// Mask value is to be applied according
+    /// to what is allowed through by the filter
+    bit_mask_filter: u64,
+    /// Mask value to be applied
+    bit_mask_value: u64,
+}
```

Comment:
Maybe rename to `RegisterValue`? The name `Modifier` is very generic, I think we need smth associated with registers.
Or we can think of this struct as a generic modifier for any future devices. Then maybe `ValueModifier`?
Also IMO `bit_mask_...` names are a bit too explicit. Just `filter` and `value` should be fine. In addition can you provide the `apply` method as a point of reference of how this masking will work, so I could do the same in my aarch64 templates?

Reply:
`filter` or `mask` and `value` look simpler to me.
Having an apply method doesn't look wrong to me, as it will reduce a room for interpretation of how filter/value should be used.

Reply:
> rename to RegisterValue

It is a modifier of a register's value, there is a difference conceptually within the code base. RegisterBitMask or RegisterModifier would be more accurate.

> filter or mask and value look simpler to me.

If the struct has a clarifying name then I can consider shortening. But the notion of being a bit mask needs to be incorporated into the naming.

Reply:
> If the struct has a clarifying name then I can consider shortening. But the notion of being a bit mask needs to be incorporated into the naming.

If we use `mask`, I think it's rather clear that it is a bit mask.

Reply:
> In short, not here no. How the CPU sub-modules make use of the data is discretionary.


If modules can use data however they want, then there is no consistency. It can be that x86 registers configured one way and aarch64 another. Isn't this the opposite of what we want? Also it should be generic over u32, u64,... because registers can have different sizes.